openmdao/util/options.py

```diff
@@ -47,35 +49,80 @@ def add_option(self, name, value, lower=None, upper=None, values=None,
             print("raising an error")
             raise ValueError("Option '{}' already exists".format(name))
 
-        self._options[name] = {
+        opt = {
             'val':    value,
-            'lower':    lower,
-            'upper':   upper,
+            'lower':  lower,
+            'upper':  upper,
             'values': values,
-            'desc' : desc,
+            'desc' :  desc,
         }
 
-        self._check(name, value)
+        self._check(name, value, opt)
+
+        self._options[name] = opt
+
+    def remove_option(self, name):
```

Comment:
To be totally strict, the functions added to the OptionsDictionary need unit tests.